### PR TITLE
yuzu-{ea,mainline}: {2901,1131} -> {2907,1137}

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Description of changes
This changes the update script to be compatible with the maintainer script in `./maintainers/scripts/update.nix`, which will hopefully make sure r-ryantm can update these packages automatically for us.

The script now also automatically updates the compatibility list, as @Docteh pointed out a mirror is available on Github. I heard it needs some work upstream and will not be updated for a while, but we might as well check for changes automatically.

The call in `all-packages.nix` is changed as well to avoid a double `callPackage`, which can make overriding quite a pain.

cc @sbruder

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and “core” functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sbruder — APPROVED** [review]
Thanks! The update script gets called from the maintainer update script and both branches worked in my test.

File: pkgs/applications/emulators/yuzu/update.sh
Comment (IvarWithoutBones) [reply pkgs/applications/emulators/yuzu/update.sh:8]: Good idea.

**Comment by IvarWithoutBones** [comment]
After taking a look at [repology](https://repology.org/projects/?search=yuzu&maintainer=&category=&inrepo=&notinrepo=&repos=&families=&repos_newest=&families_newest=) I noticed all other package repos call the early access package `yuzu-early-access`, so I've renamed ours as well. While I was at it I converted the `yuzu` alias to a throw, its quite old now.

**Review by sbruder — COMMENTED** [review]
Finally the patch is no longer needed, that’s nice :)

As for renaming the early access package, it looks like only the AUR calls it `yuzu-early-access` (though also no one besides nixpkgs calls it `yuzu-ea`), but it certainly also makes it clearer what it is.

Also, because you mentioned this in #187070: Maybe we could add `--set QT_QPA_PLATFORM xcb` to the `makeWrapper` arguments to simplify running it from Wayland (where often the platform is forced to `wayland` or `wayland-egl`)?

**Comment by IvarWithoutBones** [comment]
> As for renaming the early access package, it looks like only the AUR calls it yuzu-early-access (though also no one besides nixpkgs calls it yuzu-ea), but it certainly also makes it clearer what it is.

Yeah, I noticed after placing that comments that "all other package repositories" really doesnt hold up :sweat_smile: 

> Also, because you mentioned this in https://github.com/NixOS/nixpkgs/pull/187070: Maybe we could add --set QT_QPA_PLATFORM xcb to the makeWrapper arguments to simplify running it from Wayland (where often the platform is forced to wayland or wayland-egl)?

Good idea, thanks for the suggestion! I've added the environment variable. Also switched our `makeWrapper` arguments to `qtWrapperArgs` to avoid double wrapping.

**Review by sbruder — APPROVED** [review]
Thanks! Both branches now run without any special treatment. Also, getting rid of the double wrapping is nice.

**Comment by IvarWithoutBones** [comment]
The quick reviews are much appreciated, as always :)

If no one objects I'll go ahead and merge this in a day or two

File: pkgs/top-level/aliases.nix
Comment (adisbladis) [inline pkgs/top-level/aliases.nix:1552]: I don't see the point of throwing, it's unfriendly when we have an obvious thing to remap it to.

**Review by adisbladis — CHANGES_REQUESTED** [review]

File: pkgs/top-level/aliases.nix
Comment (IvarWithoutBones) [reply pkgs/top-level/aliases.nix:1552]: I was thinking that its just of a pointless alias, added around a year ago purely because another branch was introduced. It was added back then for backwards compatibility, and seeing how a while has passed now I figured I'd just remove it.

I don't feel strongly about this though, I'll add the alias back shortly 👍

I'm also curious; I've have seen a fair bit of other aliases being converted to a throw as well, would you consider all of those unfriendly?

Comment (IvarWithoutBones) [reply pkgs/top-level/aliases.nix]: Dropped the commit :+1:

Comment (adisbladis) [reply pkgs/top-level/aliases.nix]: > I was thinking that its just of a pointless alias, added around a year ago purely because another branch was introduced. It was added back then for backwards compatibility, and seeing how a while has passed now I figured I'd just remove it.

I only recently started using Yuzu, and I think being able to `nix-shell -p yuzu` without being aware of the different branches is worth the alias.

**Pull request merged**
